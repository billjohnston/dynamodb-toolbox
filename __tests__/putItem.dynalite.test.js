const { advanceTo } = require('jest-date-mock')
const Entity = require('../classes/Entity')
const Table = require('../classes/Table')
const createTable = require('./tables/create-table.json')
const { DocumentClient, DynamoDB } = require('../__tests__/util/dynalite')

const mockTableName = 'mock-table-put-item' 

const MyTable = new Table({
  name: mockTableName,
  partitionKey: 'pk',
  sortKey: 'sk',
  DocumentClient
})

const Customer = new Entity({
  name: 'Customer',
  attributes: {
    id: { partitionKey: true },
    sk: { hidden: true, sortKey: true },
    name: { map: 'data' },
    co: { alias: 'company' },
    age: { type: 'number' },
    status: ['sk',0],
    date_added: ['sk',1],
  },
  table: MyTable
})

  
describe('put item integration test', () => {
  beforeAll(() => DynamoDB.createTable({
    TableName: mockTableName,
    ...createTable
  }).promise())

  it('should create customer record', async () => {
    advanceTo(new Date(2020, 4, 31, 0, 0, 0))
    let item = {
      id: 123,
      name: 'Jane Smith',
      company: 'ACME',
      age: 35,
      status: 'active',
      date_added: '2020-04-24'
    }
    
    await Customer.put(item)
    const scan = await DocumentClient.scan({
        TableName: mockTableName
    }).promise()
    expect(scan).toMatchSnapshot()
  })

})