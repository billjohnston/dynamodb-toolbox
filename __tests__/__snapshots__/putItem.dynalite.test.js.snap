// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`put item integration test should create customer record 1`] = `
Object {
  "Count": 1,
  "Items": Array [
    Object {
      "_ct": "2020-05-31T00:00:00.000Z",
      "_et": "Customer",
      "_md": "2020-05-31T00:00:00.000Z",
      "age": 35,
      "co": "ACME",
      "data": "Jane Smith",
      "date_added": "2020-04-24",
      "pk": "123",
      "sk": "active#2020-04-24",
      "status": "active",
    },
  ],
  "ScannedCount": 1,
}
`;
